packages/site/src/preview/useAtlantisPreviewViewer.tsx

```diff
@@ -1,23 +1,63 @@
 import { useRef, useState } from "react";
+import { ComponentType } from "../types/content";
+import { getPlatformForComponentType } from "../utils/componentTypeUtils";
 
 export const useAtlantisPreviewViewer = () => {
   const iframe = useRef<HTMLIFrameElement>(null);
   const iframeMobile = useRef<HTMLIFrameElement>(null);
 
-  const [type, setType] = useState<"web" | "mobile">(
-    new URLSearchParams(window.location.search)?.has("mobile")
-      ? "mobile"
-      : "web",
-  );
+  // Parse the current type from URL parameters
+  const getInitialType = (): ComponentType => {
+    const params = new URLSearchParams(window.location.search);
+    if (params.has("mobile")) return "mobile";
+    if (params.has("webLegacy")) return "web";
```

Comment:
not sure if I love the "webLegacy" URL param

main reason I went with that is because it leaves the door open to "mobileLegacy" if that's ever a thing. plus it's just a URL, not a very important thing.